tests/test_async_tx.py

```diff
@@ -104,3 +108,54 @@ async def test_async_transaction_exclusive(self):
                 ):
                     await asyncio.wait_for(f1, timeout=5)
                     await asyncio.wait_for(f2, timeout=5)
+
+    async def test_async_transaction_savepoint_1(self):
+        async for tx in self.client.transaction():
+            async with tx:
+                sp1 = await tx.declare_savepoint("sp1")
+                sp2 = await tx.declare_savepoint("sp2")
+                with self.assertRaisesRegex(
+                    edgedb.InterfaceError, "savepoint.*already exists"
+                ):
+                    await tx.declare_savepoint("sp1")
```

Comment:
This is contrary to server savepoint semantics, masking an earlier savepoint is allowed:

```
edgedb> start transaction;
OK: START TRANSACTION
edgedb[tx]> declare savepoint sp1;
OK: DECLARE SAVEPOINT
edgedb[tx]> declare savepoint sp1;
OK: DECLARE SAVEPOINT
```

Reply:
This releases previous savepoint with that name, right?

Reply:
Yeah, I think Paul is right.